app/components/Home.tsx

```diff
@@ -0,0 +1,270 @@
+import * as React from 'react';
+const { EditorKit, EditorKitDelegate } = require('sn-editor-kit');
+const MarkdownIt = require('markdown-it');
+
+const EditMode = 0;
+const SplitMode = 1;
+const PreviewMode = 2;
+
+const modes = [
+  { mode: EditMode, label: "Edit", css: "edit" },
+  { mode: SplitMode, label: "Split", css: "split" },
+  { mode: PreviewMode, label: "Preview", css: "preview" },
+];
+
+const markdownitOptions = {
+  // automatically render raw links as anchors.
+  linkify: true
+};
+
+const MarkdownParser = MarkdownIt(markdownitOptions)
+  .use(require('markdown-it-footnote'))
+  .use(require('markdown-it-task-lists'))
+  .use(require('markdown-it-highlightjs'));
+
+type AppProps = {
+  text?: string,
+  mode: any;
+  platform?: any,
+}
+
+const initialState = {
+  mode: modes[0],
+  text: "",
+}
+const debugMode = false;
+
+if (debugMode) {
+  console.log("home loaded 2");
+}
+let keyMap = new Map();
+
+export class Home extends React.Component<{}, AppProps> {
+  editorKit: any;
+  note: any;
+
+  constructor(props: AppProps) {
+    super(props);
+    if (debugMode) {
+      console.log("started");
+    }
+    this.state = initialState;
+  }
+
+  componentDidMount = () => {
+    this.configureEditorKit();
+    this.configureMarkdown();
+    this.configureResizer();
+  }
+
+  configureEditorKit() {
+    if (debugMode){
+      console.log("editor kit loaded");
+    }
+    let delegate = new EditorKitDelegate({
+      setEditorRawText: (text: string) => {
+        this.setState(
+          {
+            text,
+          }, () => {
+            this.updatePreviewText();
+            this.loadSavedMode();
+          });
+      },
+      clearUndoHistory: () => { },
+      //getElementsBySelector: () => [],
+    });
+
+    this.editorKit = new EditorKit({
+      delegate: delegate,
+      mode: 'plaintext',
+      supportsFilesafe: false,
+    });
+  };
+
+  saveNote = () => {
+    this.editorKit.onEditorValueChanged(this.state.text);
+  }
+
+  handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const target = event.target;
+    const value = target.value;
+
+    this.setState({
+      text: value
+    }, () => {
+      this.updatePreviewText();
+      this.saveNote();
+    });
+  };
+
+  loadSavedMode = () => {
+    try {
+      let savedMode = this.editorKit.internal.componentManager.componentDataValueForKey("mode");
+      if (savedMode) {
+        this.setModeFromModeValue(savedMode);
+      }
+      this.setState({ platform: this.editorKit.internal.componentManager.platform });
+    }
+    finally {
+      return
```

Comment:
Empty return not necessary.

Reply:
Not having `finally` gives a `'finally' expected.ts(1005)` error in VS Code

Reply:
You could have an empty catch `statement` I suppose.

Reply:
Fixed with https://github.com/sn-extensions/markdown-basic/pull/29/commits/5be724055c9b0acce041b788788c6e4067336911